### PR TITLE
Add import warnings for soiling and availability modules

### PR DESCRIPTION
**Pull request by kanderso-nrel**
- ~Code changes are covered by tests~
- [x] New functions added to `__init__.py`
- [x] API.rst is up to date, along with other sphinx docs pages
- [x] Example notebooks are rerun and differences in results scrutinized
- [x] Updated changelog

Behavior:

```
In [1]: import rdtools

In [2]: from rdtools.soiling import SRRAnalysis
C:\Users\KANDERSO\projects\rdtools\rdtools\soiling.py:9: UserWarning: The soiling analysis functions are currently experimental. The API and results may change in future releases as the code matures.
  'The soiling analysis functions are currently experimental. The API and '

In [3]: from rdtools.availability import AvailabilityAnalysis
C:\Users\KANDERSO\projects\rdtools\rdtools\availability.py:14: UserWarning: The availability analysis functions are currently experimental. The API and results may change in future releases as the code matures.
  'The availability analysis functions are currently experimental. The API '
```

### CODE REVIEW

File: rdtools/__init__.py [hunk 1]
```diff
@@ -14,15 +14,15 @@
...
-from rdtools.soiling import monthly_soiling_rates
-from rdtools.soiling import annual_soiling_ratios
+# from rdtools.soiling import soiling_srr
+# from rdtools.soiling import monthly_soiling_rates
+# from rdtools.soiling import annual_soiling_ratios
 from rdtools.plotting import degradation_summary_plots
 from rdtools.plotting import soiling_monte_carlo_plot
 from rdtools.plotting import soiling_interval_plot
 from rdtools.plotting import soiling_rate_histogram
 from rdtools.plotting import availability_summary_plots
```
Comment (kanderso-nrel) [inline rdtools/__init__.py:24]: Should we comment out the soiling/availability plotting function imports too?  Do they need a warning?

File: rdtools/availability.py [hunk 2]
```diff
@@ -10,6 +10,11 @@
 from scipy.interpolate import interp1d
 import warnings
 
+warnings.warn(
+    'The availability analysis functions are currently experimental. The API '
+    'and results may change in future releases as the code matures.'
```
Comment (mdeceglie) [inline rdtools/availability.py:15]: What do you think about:
```
The availability module is currently experimental. The API, results,
and default behaviors may change in future releases (including MINOR
and PATCH releases) as the code matures.
```

File: rdtools/soiling.py [hunk 3]
```diff
@@ -5,6 +5,10 @@
 import numpy as np
 from scipy.stats.mstats import theilslopes
 
+warnings.warn(
+    'The soiling analysis functions are currently experimental. The API and '
```
Comment (mdeceglie) [inline rdtools/soiling.py:9]: (similar suggestion as in availability)

**Comment by mdeceglie** [comment]
Thanks for getting this started @kanderso-nrel. I wonder if we should do something in RTD as well. Perhaps in the overview and or in the API section (maybe via module-level docstring). A more involved option would be to break out a separate part in the RTD sidebar for "API (Experimental)" or similar. We should seek a balance between making the experimental nature obvious to folks and giving ourselves too many things to remember to change back when these solidify.

**Comment by kanderso-nrel** [comment]
Rather than hardcoding (slightly different) module descriptions in `api.rst`, the last commit uses the `automodule` directive to bring in the module docstrings.  Because of the way `automodule` affects downstream imports, it was easiest to change the function lists to be just the function name instead of module.function, e.g. they show as `clip_filter` instead of `filtering.clip_filter` in the function lists.  Example: https://rdtools--213.org.readthedocs.build/en/213/api.html#soiling

File: rdtools/__init__.py (hunk 1 above)
Comment (mdeceglie) [reply rdtools/__init__.py:24]: I'm leading toward commenting them out of `__init__.py`, adding a warning at the function level, and adding a note to the function docstrings.

Comment (kanderso-nrel) [reply rdtools/__init__.py]: Done, example: https://rdtools--213.org.readthedocs.build/en/213/generated/rdtools.plotting.soiling_interval_plot.html

**Review by mdeceglie — APPROVED** [review]
Looks good, thanks @kanderso-nrel

**Pull request merged**